pkg/main/main.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strconv"
+	"strings"
+)
+
+const serversEnvVar = "SERVERS"
+const serverCmdEnvVar = "SERVER_CMD"
+
+// Constants indicating the types of errors for Dispatch function invocation
+const (
+	InputError    = "InputError"
+	FunctionError = "FunctionError"
+	SystemError   = "SystemError"
+)
+
+var servers []Server
+var router Router
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	reader := r.Body
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(reader)
+	rawBody := buf.Bytes()
+	var body map[string]interface{}
+	err := json.Unmarshal(rawBody, &body)
+	if err != nil {
+		resp := Response{
+			Context: Context{
+				Error: Error{
+					ErrorType: InputError,
+					Message:   "Invalid Input",
+				},
+			},
+		}
+		out, _ := json.Marshal(resp)
+		fmt.Fprintf(w, string(out))
+		return
+	}
+
+	resp, _ := router.Delegate(body)
```

Comment:
using a non-global router would make it easier to test this handler. you could create a closure that takes router and returns the handler. Or define a small `Proxy` struct type that keeps router as an internal field (and implements `ServeHTTP` method).